api/pkg/apis/v1alpha1/providers/stage/wait/wait.go

```diff
@@ -161,13 +162,13 @@ func (i *WaitStageProvider) Process(ctx context.Context, mgrContext contexts.Man
 			prefixedNames[i] = fmt.Sprintf("%v-%v", inputs["__origin"], object)
 		}
 	}
+	namespace := stage.ReadInputString(inputs, "__namespace")
+	if namespace == "" {
+		namespace = "default"
+	}
 
-	log.Debugf("  P (Wait Processor): waiting for %v %v", objectType, prefixedNames)
+	log.Debugf("  P (Wait Processor): waiting for %v %v in namespace %s", objectType, prefixedNames, namespace)
 	counter := 0
-	namespace := "default"
-	if s, ok := inputs["namespace"]; ok {
```

Comment:
I think we should use GetNamespace() here as well so that the wait stage can wait for objects from given namespace

Reply:
We don't need to change here?

Reply:
I have already changed it to GetNamespace here.